src/check_result.jl

```diff
@@ -25,6 +25,10 @@ for (T1, T2) in ((AbstractThunk, Any), (AbstractThunk, AbstractThunk), (Any, Abs
     end
 end
 
+check_equal(::Zero, x; kwargs...) = check_equal(zero(x), x; kwargs...)
+check_equal(x, ::Zero; kwargs...) = check_equal(x, zero(x); kwargs...)
+check_equal(x::Zero, y::Zero; kwargs...) = @test true
```

Comment:
No real objections to having this, but does it ever occur? I though `check_equal` was for checking AD vs FD results, so not sure when FD would give `Zero()`? Or is it ever used outside this context?

Reply:
yes, this is used in some other places.
it checks primal results,
and it is used to check `add!!(x,y) vs `x+y`
and in that later case exactly this can indeed occur